src/ppc/src/import/problem_importer.rs

```diff
@@ -0,0 +1,392 @@
+use anyhow::Context as _;
+use std::{collections::HashSet, path::Path};
+
+pub(crate) struct Importer<'a> {
+    pub(crate) src: &'a Path,
+    pub(crate) dest: &'a Path,
+    pub(crate) problem_cfg: crate::manifest::RawProblem,
+    pub(crate) known_generators: HashSet<String>,
+    pub(crate) doc: roxmltree::Node<'a, 'a>,
+    pub(crate) limits: pom::Limits,
+}
+
+enum FileCategory {
+    Validator,
+    Checker,
+    Generator,
+}
+
+impl FileCategory {
+    fn derive(name: &str) -> Option<FileCategory> {
+        if name == "check" || name == "checker" {
+            return Some(FileCategory::Checker);
+        }
+
+        if name == "validator" {
+            return Some(FileCategory::Validator);
+        }
+
+        if name.starts_with("gen") {
+            return Some(FileCategory::Generator);
+        }
+
+        None
+    }
+}
+
+impl<'a> Importer<'a> {
+    // <problem><judging> is most important section for us: it contains information
+    // about tests
+    fn process_judging_section(&mut self, node_judging: roxmltree::Node) -> anyhow::Result<()> {
+        let node_testset = node_judging.first_element_child().unwrap();
+        assert_eq!(node_testset.tag_name().name(), "testset");
+
+        let mut test_pattern = None;
+        let mut ans_pattern = None;
+        for child in node_testset.children() {
+            if !child.is_element() {
+                continue;
+            }
+            match child.tag_name().name() {
+                "time-limit" => {
+                    let tl = child
+                        .text()
+                        .unwrap()
+                        .parse::<u64>()
+                        .expect("parsing <time-limit>:");
+                    println!("time limit: {} ms", tl);
```

Comment:
Is it a debug message?

Reply:
No, it isn't.
I think ppc is too verbose, but it will be solved some other day.